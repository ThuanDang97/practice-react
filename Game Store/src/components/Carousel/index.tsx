import Carousel from 'react-material-ui-carousel'
import { Box, CardMedia, Paper, useTheme } from '@mui/material'

export interface ICarousel {
  pathImages: {
    pathImage: string
  }[]
}

const Carousell = ({ pathImages }: ICarousel) => {
  const theme = useTheme()
  return (
    <Box minWidth={650} maxWidth={1800} width="100%">
      <Carousel
        animation="slide"
        navButtonsAlwaysVisible={true}
        navButtonsProps={{
          style: {
            backgroundColor: 'transparent',
            color: theme.palette.common.white,
            borderRadius: 5,
          },
        }}
        activeIndicatorIconButtonProps={{
          style: {
            color: theme.palette.secondary.dark,
            fontSize: 20,
          },
        }}
        indicatorIconButtonProps={{
          style: {
            padding: '5px',
          },
        }}
      >
        {pathImages.map((pathItem, index) => (
          <Item
            key={`Carousel + ${index.toString()}`}
            pathImage={pathItem.pathImage}
          />
        ))}
      </Carousel>
    </Box>
  )
}

const Item = ({ pathImage }: { pathImage: string }) => {
  return (
    <Paper>
      <CardMedia
        sx={{ borderRadius: 1, maxHeight: '1000px', minHeight: '760px' }}
        component="img"
        image={pathImage}
        alt="green iguana"
      />
    </Paper>
  )
}

export default Carousell
