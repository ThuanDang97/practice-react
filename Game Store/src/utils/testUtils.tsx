import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@mui/material/styles'
import { BrowserRouter as Router } from 'react-router-dom'

// Themes
import theme from '@self-types/Themes'

// Contexts
import { AuthProvider } from '@contexts/AuthProvider'
import { CartProvider } from '@contexts/CartProvider'

export const customRender = (component: ReactNode) => {
  return render(
    <ThemeProvider theme={theme}>
      <AuthProvider>
        <CartProvider>
          <Router>{component}</Router>
        </CartProvider>
      </AuthProvider>
    </ThemeProvider>,
  )
}

export * from '@testing-library/react'
export { customRender as render }
