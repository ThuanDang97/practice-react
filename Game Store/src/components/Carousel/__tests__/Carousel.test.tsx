// Utils
import { render } from '@utils/testUtils'

// Components
import Carousel from '..'

describe('Carousel component', () => {
  const MOCK_CAROUSEL = [
    {
      pathImage:
        'https://media.rawg.io/media/games/456/456dea5e1c7e3cd07060c14e96612001.jpg',
    },
    {
      pathImage:
        'https://media.rawg.io/media/games/456/456dea5e1c7e3cd07060c14e96612001.jpg',
    },
    {
      pathImage:
        'https://media.rawg.io/media/games/456/456dea5e1c7e3cd07060c14e96612001.jpg',
    },
    {
      pathImage:
        'https://media.rawg.io/media/games/456/456dea5e1c7e3cd07060c14e96612001.jpg',
    },
    {
      pathImage:
        'https://media.rawg.io/media/games/456/456dea5e1c7e3cd07060c14e96612001.jpg',
    },
  ]

  it('should render Card Detail', () => {
    const { container } = render(<Carousel pathImages={MOCK_CAROUSEL} />)

    expect(container).toMatchSnapshot()
  })
})
