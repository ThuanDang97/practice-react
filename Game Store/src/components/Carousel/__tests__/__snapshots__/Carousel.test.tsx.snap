// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel component should render Card Detail 1`] = `
<div>
  <div
    class="MuiBox-root css-4493d5"
  >
    <div
      class=" css-ohwg9z"
    >
      <div
        class="css-1f8sh1y"
      >
        <div
          class="css-1l7c0cy"
        >
          <div
            draggable="false"
            style="height: 100%; user-select: none; transform: translate3d(0px, 0px, 0) scale(NaN, NaN); transform-origin: 50% 50% 0;"
          >
            <div
              style="position: relative; height: 100%; opacity: 0; z-index: 1; transform: none;"
            >
              <div>
                <div
                  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
                >
                  <img
                    alt="green iguana"
                    class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-ajwyql-MuiCardMedia-root"
                    src="https://media.rawg.io/media/games/456/456dea5e1c7e3cd07060c14e96612001.jpg"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-1l7c0cy"
        >
          <div
            draggable="false"
            style="height: 100%; user-select: none; transform: translate3d(0px, 0px, 0) scale(NaN, NaN); transform-origin: 50% 50% 0;"
          >
            <div
              style="position: relative; height: 100%; opacity: 0; display: none; z-index: 0; transform: translateX(0%) translateZ(0); z-index: 0;"
            >
              <div>
                <div
                  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
                >
                  <img
                    alt="green iguana"
                    class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-ajwyql-MuiCardMedia-root"
                    src="https://media.rawg.io/media/games/456/456dea5e1c7e3cd07060c14e96612001.jpg"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-1l7c0cy"
        >
          <div
            draggable="false"
            style="height: 100%; user-select: none; transform: translate3d(0px, 0px, 0) scale(NaN, NaN); transform-origin: 50% 50% 0;"
          >
            <div
              style="position: relative; height: 100%; opacity: 0; display: none; z-index: 0; transform: translateX(0%) translateZ(0); z-index: 0;"
            >
              <div>
                <div
                  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
                >
                  <img
                    alt="green iguana"
                    class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-ajwyql-MuiCardMedia-root"
                    src="https://media.rawg.io/media/games/456/456dea5e1c7e3cd07060c14e96612001.jpg"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-1l7c0cy"
        >
          <div
            draggable="false"
            style="height: 100%; user-select: none; transform: translate3d(0px, 0px, 0) scale(NaN, NaN); transform-origin: 50% 50% 0;"
          >
            <div
              style="position: relative; height: 100%; opacity: 0; display: none; z-index: 0; transform: translateX(0%) translateZ(0); z-index: 0;"
            >
              <div>
                <div
                  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
                >
                  <img
                    alt="green iguana"
                    class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-ajwyql-MuiCardMedia-root"
                    src="https://media.rawg.io/media/games/456/456dea5e1c7e3cd07060c14e96612001.jpg"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-1l7c0cy"
        >
          <div
            draggable="false"
            style="height: 100%; user-select: none; transform: translate3d(0px, 0px, 0) scale(NaN, NaN); transform-origin: 50% 50% 0;"
          >
            <div
              style="position: relative; height: 100%; opacity: 0; display: none; z-index: 0; transform: translateX(0%) translateZ(0); z-index: 0;"
            >
              <div>
                <div
                  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
                >
                  <img
                    alt="green iguana"
                    class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-ajwyql-MuiCardMedia-root"
                    src="https://media.rawg.io/media/games/456/456dea5e1c7e3cd07060c14e96612001.jpg"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class=" css-1abc02a"
      >
        <button
          aria-label="Next"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium  css-1g04gm2-MuiButtonBase-root-MuiIconButton-root"
          style="background-color: transparent; color: rgb(255, 255, 255); border-radius: 5px;"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="NavigateNextIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class=" css-hn784z"
      >
        <button
          aria-label="Previous"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium  css-1g04gm2-MuiButtonBase-root-MuiIconButton-root"
          style="background-color: transparent; color: rgb(255, 255, 255); border-radius: 5px;"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="NavigateBeforeIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class=" css-1m9128y"
      >
        <button
          aria-label="carousel indicator 1"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium   css-8ucetn-MuiButtonBase-root-MuiIconButton-root"
          style="padding: 5px; color: rgb(123, 31, 162); font-size: 20px;"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-aq8pb7-MuiSvgIcon-root"
            data-testid="FiberManualRecordIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="8"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="carousel indicator 2"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium  css-bbreus-MuiButtonBase-root-MuiIconButton-root"
          style="padding: 5px;"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-aq8pb7-MuiSvgIcon-root"
            data-testid="FiberManualRecordIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="8"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="carousel indicator 3"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium  css-bbreus-MuiButtonBase-root-MuiIconButton-root"
          style="padding: 5px;"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-aq8pb7-MuiSvgIcon-root"
            data-testid="FiberManualRecordIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="8"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="carousel indicator 4"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium  css-bbreus-MuiButtonBase-root-MuiIconButton-root"
          style="padding: 5px;"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-aq8pb7-MuiSvgIcon-root"
            data-testid="FiberManualRecordIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="8"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="carousel indicator 5"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium  css-bbreus-MuiButtonBase-root-MuiIconButton-root"
          style="padding: 5px;"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-aq8pb7-MuiSvgIcon-root"
            data-testid="FiberManualRecordIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="8"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
